test_runner/src/main/kotlin/ftl/args/ValidateCommonArgs.kt

```diff
@@ -8,13 +8,27 @@ fun CommonArgs.validate() {
     assertShardTime()
     assertRepeatTests()
     assertSmartFlankGcsPath()
+    assertOrientationCorrectness()
+}
+
+private fun CommonArgs.assertOrientationCorrectness() {
+    val orientationsList = listOf("portrait", "landscape", "default")
+    val mispeltOrientations = mutableListOf<String>()
+    devices.map { it.orientation }.forEach {
```

Comment:
Maybe we can use filter instead of forEach and mutableList?

```kotlin

    val mispeltOrientations = devices.map { it.orientation }.filter { it !in orientationsList }

```

or more functional

```kotlin


private fun CommonArgs.assertOrientationCorrectness() = devices.deviceWithMisspelledOrientation(listOf("portrait", "landscape", "default")).throwIfAnyMisspelled()

private fun List<Device>.deviceWithMisspelledOrientation(availableOrientations: List<String>) = filter { it.orientation !in availableOrientations }.map { it.orientation }

private fun List<String>.throwIfAnyMisspelled() =
    if (isNotEmpty()) throw FlankConfigurationError("Orientation misspelled or incorrect, found ${joinToString()}. Aborting.")
    else Unit

```

But this only my idea. WDYT?

Reply:
I think for now i will rewrite it to the more functional way just because the project is focused this way and its good practise. :) else its such a small section i am not sure of its worth besides that.